server/commands/groupCreator.ts

```diff
@@ -0,0 +1,39 @@
+import type { Transaction } from "sequelize";
+import { Event, Group, type User } from "@server/models";
+
+type Props = {
+  name: string;
+  actor: User;
+  ip: string;
+  transaction?: Transaction;
+};
+
+export default async function groupCreator({
+  name,
+  actor,
+  ip,
+  transaction,
+}: Props): Promise<Group> {
+  const group = await Group.create(
+    {
+      name,
+      teamId: actor.teamId,
+      createdById: actor.id,
+    },
+    { transaction }
+  );
+  await Event.create(
```

Comment:
We should retain all these as `Event.createFromContext` to automatically add the correct actor, team, and ip.

Reply:
The other existing commands do not take the context in input and rely on `Event.create`.
I think there may be some cases where you want to call these commands but an `APIContext` is not available (that what I'm doing for OIDC group mapping).

Reply:
It is unfortunate that `accountProvisioner` does not already get passed `ctx` but it can/should and that would solve the issue

Reply:
This `ctx` would be a simple `Context`, not a complete `APIContext`, since the user hasn't been created yet and doesn't have an access token. I wouldn't be able to properly create an `APIContext` to pass to these commands. 🫤